src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,82 @@
+package roomescape.service;
+
+import org.springframework.stereotype.Service;
+import roomescape.dao.ReservationDao;
+import roomescape.dao.ThemeDao;
+import roomescape.dao.TimeDao;
+import roomescape.domain.reservation.Reservation;
+import roomescape.domain.theme.Theme;
+import roomescape.domain.time.Time;
+import roomescape.dto.reservation.ReservationRequest;
+import roomescape.dto.reservation.ReservationResponse;
+import roomescape.dto.reservation.ReservationTimeInfosResponse;
+import roomescape.dto.reservation.ReservationsResponse;
+import roomescape.global.exception.error.ErrorType;
+import roomescape.global.exception.model.ConflictException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+
+@Service
+public class ReservationService {
+
+    private final ReservationDao reservationDao;
+    private final TimeDao timeDao;
+    private final ThemeDao themeDao;
+
+    public ReservationService(final ReservationDao reservationDao,
+                              final TimeDao timeDao,
+                              final ThemeDao themeDao) {
+        this.reservationDao = reservationDao;
+        this.timeDao = timeDao;
+        this.themeDao = themeDao;
+    }
+
+    public ReservationsResponse findAllReservations() {
+        List<ReservationResponse> response = reservationDao.findAll()
+                .stream()
+                .map(ReservationResponse::from)
+                .toList();
+
+        return new ReservationsResponse(response);
+    }
+
+    public ReservationTimeInfosResponse findReservationsByDateAndThemeId(final LocalDate date, final Long themeId) {
+        return timeDao.findByDateAndThemeId(date, themeId);
+    }
+
+    public ReservationResponse addReservation(final ReservationRequest reservationRequest) {
+        LocalDate today = LocalDate.now();
+        LocalDate requestDate = reservationRequest.date();
+        Time time = timeDao.findById(reservationRequest.timeId());
+        Theme theme = themeDao.findById(reservationRequest.themeId());
+
+        validateDateAndTime(requestDate, today, time);
+        validateReservationDuplicate(reservationRequest, theme);
+
+        Reservation savedReservation = reservationDao.insert(reservationRequest.toReservation(time, theme));
+
+        return ReservationResponse.from(savedReservation);
+    }
+
+    private void validateDateAndTime(final LocalDate requestDate, final LocalDate today, final Time time) {
+        if (requestDate.isBefore(today) || (requestDate.isEqual(today) && time.getStartAt()
+                .isBefore(LocalTime.now()))) {
```

Comment:
내부 조건식 가독성이 떨어져요.
깨진 유리창 효과 때문에 이런식으로 조건식을 만들어두면 다른 개발자들이 계속 || 와 괄호로 감싸진 조건식을 덧대게 됩니다

`requestDate.isBefore(today)`와 `(requestDate.isEqual(today) && time.getStartAt()
                .isBefore(LocalTime.now())`는 같은 추상화 수준이니 뒤 조건식을 메서드로 추출해서 더 쉽게 파악되도록 해주세요
```suggestion
        if (requestDate.isBefore(today) || 
            something(today, time)) {
```

Reply:
```java
    private boolean isReservationInPast(final LocalDate requestDate, final Time requestTime, final LocalDateTime now) {
        LocalDate today = now.toLocalDate();
        LocalTime nowTime = now.toLocalTime();

        // 날짜가 지난 경우
        if (requestDate.isBefore(today)) {
            return true;
        }
        // 날짜가 오늘이지만, 지난 시간인 경우
        if (requestDate.isEqual(today) && requestTime.getStartAt().isBefore(nowTime)) {
            return true;
        }
        return false;
    }
```
혹시 검증 메서드의 이름을 지난 날짜에 대한 예약인 지 검증한다는 뜻의 isReservationInPast 로 설정하여 같은 추상화 수준을 만들어주고
위와 같이 조건문을 2개로 분리후 주석으로 의미를 정확히 표기해주는 것은 어떤가요??
필요하다면 메서드를 분리해주는 것도 좋지만, 같은 추상화 수준으로 묶어줄 수 있을 것 같아서요!

Reply:
네네 이것도 좋은 방법이라고 생각합니다

대신 주석은 필요없다고 생각합니다 ㅎ 
코드와 동어 반복인 주석은 공해입니다 (주석도 유지보수 대상인데 둘이 달라지면 더 헷갈리겠죠 보통 주석을 믿거든요)

주석에 대해선 제가 뭐라고 하는 것 보다 우테코에 많다고 알고있는 [좋은코드 나쁜코드](https://www.yes24.com/Product/Goods/109366833) 5.2 주석문의 적절한 사용 (127p) 를 봐주세용

Reply:
동어 반복 + 로직 변경 시 주석도 변경을 잊어버리게 되면 문제가 생긴다는 점에서 제거하는 편이 좋겠네요!
좋은 의견 감사합니다.
주석 처리하고 내일 캠퍼스에 가면 책도 바로 읽어보겠습니다